### PR TITLE
Add mask combination method

### PR DESCRIPTION
**Pull request by renan028**
# Description
part of: [AB#17228](https://dev.azure.com/rapyuta-robotics/flappter/_workitems/edit/17228)
- Instead of using flag `use_maximum_`, use the combination method
- Add a new combination method for static layer: `Mask`

The idea is to be able to subtract the static layer from the costmap, so that we use only the obstacle layer, without the static obstacles (`map`). Video below.

https://github.com/user-attachments/assets/97a552fd-d310-40de-8a30-bc86a9f19184

Note that the plugins order matter. In this case, we should use:
```
plugins:
  - {name: obstacle,        type: "costmap_2d::ObstacleLayer"}
  - {name: static,          type: "costmap_2d::StaticLayer"}
```

### CODE REVIEW

File: costmap_2d/cfg/StaticPlugin.cfg [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, bool_t, double_t, int_t
+
+gen = ParameterGenerator()
+
+gen.add("enabled", bool_t, 0, "Whether to apply this plugin or not", True)
+
+combo_enum = gen.enum([gen.const("Maximum", int_t,  0, "Take the maximum of the values"),
+                       gen.const("Overwrite", int_t,  1, "Overwrite values"),
+                       gen.const("Subtraction", int_t,  2, "LETHAL becomes FREE")],
```
Comment (corot) [inline costmap_2d/cfg/StaticPlugin.cfg]: S: Subtract?

**Review by tomin11 — APPROVED** [review]

**Review by plsang — APPROVED** [review]

**Comment by corot** [comment]
@renan028 do we still need this?

**Comment by renan028** [comment]
> @renan028 do we still need this?

I will change this to `Mask` and fix because it changes to the whole map.
Changing to draft

File: costmap_2d/include/costmap_2d/static_layer.h [hunk 2]
```diff
@@ -46,10 +46,18 @@
...
+#include <costmap_2d/GenericPluginConfig.h>
 
 namespace costmap_2d
 {
 
+enum class CombinationMethod
+{
+  MAXIMUM = 0,
+  OVERWRITE = 1,
+  MASK = 2
```
Comment (tomin11) [inline costmap_2d/include/costmap_2d/static_layer.h]: F: official [costmap](https://github.com/ros-planning/navigation/blob/9ad644198e132d0e950579a3bc72c29da46e60b0/costmap_2d/plugins/obstacle_layer.cpp#L436) uses _Overwrite" (0), "Maximum" (1)_
we could use the same

**Pull request merged**